core/src/ancestor_hashes_service.rs

```diff
@@ -1373,13 +1377,15 @@ mod test {
         blockstore
             .insert_shreds(shreds, None, false)
             .expect("Expect successful ledger write");
-        for duplicate_confirmed_slot in 0..dead_slot {
+        for duplicate_confirmed_slot in 0..(dead_slot - 1) {
             let bank_hash = correct_bank_hashes
                 .get(&duplicate_confirmed_slot)
                 .cloned()
                 .unwrap_or_else(Hash::new_unique);
             blockstore.insert_bank_hash(duplicate_confirmed_slot, bank_hash, true);
         }
+        // Insert wrong hash for `dead_slot - 1`
```

Comment:
If we don't do this then we don't get a decision, because `ancestor_hashes_service` doesn't dump the very last slot (as it should be dumped through normal dump & replay)